### PR TITLE
Sign MSAL with snk containing public and private keys

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: msal/build/strong_signing.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+### Strong-naming general notes
+
+1. Strong naming is NOT a security feature
```
Comment (jmprieur) [inline msal/build/strong_signing.md]: We might want to add a link to the docs.ms article: https://docs.microsoft.com/en-us/dotnet/framework/app-domains/strong-named-assemblies

File: msal/build/strong_signing.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+2. Signing and strong naming are orthogonal. Signing (e.g. with Authenticode) is a security feature.
+3. Changing the strong naming key on a library is discouraged because the public key becomes part of the name of the assembly. The name of an assembly is used in many configuration blocks, for example assmbly binding and assembly redirects.
```
Comment (jmprieur) [inline msal/build/strong_signing.md]: assmbly => assembly

File: msal/samples/desktop/SampleApp/SampleApp.csproj [hunk 3]
```diff
@@ -34,6 +34,10 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Appveyor|AnyCPU'">
     <OutputPath>bin\Appveyor\</OutputPath>
   </PropertyGroup>
+  <PropertyGroup>
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>../../../build/MSAL.snk</AssemblyOriginatorKeyFile>
```
Comment (jmprieur) [inline msal/samples/desktop/SampleApp/SampleApp.csproj:39]: Do we need to sign the sample apps?

File: msal/tests/Test.MSAL.NET.UIAutomation/Test.MSAL.NET.UIAutomation.csproj [hunk 4]
```diff
@@ -9,6 +9,10 @@
     <AssemblyName>Test.MSAL.NET.UIAutomation</AssemblyName>
     <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
     <TargetFrameworkProfile />
+  </PropertyGroup>
+    <PropertyGroup>
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>../../build/MSAL.snk</AssemblyOriginatorKeyFile>
```
Comment (jmprieur) [inline msal/tests/Test.MSAL.NET.UIAutomation/Test.MSAL.NET.UIAutomation.csproj:15]: and the tests? is it for the internals visible to?

**Review by jmprieur — APPROVED** [review]
LGTM.
1 typo and 1 suggestion

File: msal/build/strong_signing.md (hunk 1 above)
Comment (henrik-me) [reply msal/build/strong_signing.md]: good point

---
In reply to: [212448834](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1221#discussion_r212448834) [](ancestors = 212448834)

**Comment by henrik-me** [comment]
is this the "original" msal.snk?

File: msal/build/strong_signing.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+### Strong-naming general notes
```
Comment (henrik-me) [inline msal/build/strong_signing.md]: >rong-naming general notes [](start = 6, length = 25)

also, the same info should be in the wiki or we should link to the md from the wiki... I suppose I prefer it being in the wiki over md files in general. then here we can reference the wiki page?

**Review by henrik-me — APPROVED** [review]
:shipit:

**Comment by henrik-me** [comment]
seems like failing utests, new run here: https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=178203&_a=summary&view=logs

**Comment by henrik-me** [comment]
added a few minor modifications to the build and test

**Comment by bgavrilMS** [comment]
Thanks for the fix @henrik-me; I confirm the MSAL key is the old one used to strong-name 1.1.4-preview and older; made minor changes to the docs and moved them to the wiki.

**Pull request merged**